auth-service/src/handlers/AuthService.ts

```diff
@@ -170,41 +164,67 @@ class AuthHandler implements IAuthServer {
+    ): Promise<void> => {
         const accessToken = call.request.getAccessToken();
-
+        const response = new AccountResponse();
         // This could be done synchronously as well
-        jwt.verify(
-            accessToken,
-            config.accessToken.secret,
-            (error, decoded: any) => {
-                if (error) {
-                    console.error(error);
-                    callback(null, new AccountInfo());
-                    return;
-                }
+        try {
+            const payload = await verifyToken(accessToken, config.accessToken.secret);
+            const account = await Account.findByPk(payload.sub);
+            if (!account) {
+                response.setError(AuthErrorStatus.NOT_FOUND);
+            } else {
+                const accountInfo = accountToAccountInfo(account);
+                response.setAccount(accountInfo);
+            }
+        } catch (e) {
+            switch (e.name) {
+                case "JsonWebTokenError":
+                    response.setError(AuthErrorStatus.INVALID_TOKEN);
+                    break;
+                case "TokenExpiredError":
+                    response.setError(AuthErrorStatus.EXPIRED_TOKEN);
+                    break;
+                default:
+                    console.error("Error verifying access token:", e);
+                    response.setError(AuthErrorStatus.SERVER_ERROR);
+            }
+        }
+        return callback(null, response);
+    }
 
-                Account.findByPk(decoded.sub)
-                    .then((account) => {
-                        const response = new AccountInfo();
-                        if (!account) {
-                            return callback(null, response);
-                        }
-                        response.setId(account.id);
-                        response.setUsername(account.username);
-                        response.setCreatedAt(account.createdAt.toJSON());
-                        response.setUpdatedAt(account.updatedAt.toJSON());
-                        callback(null, response);
-                    })
-                    .catch((err) => {
-                        console.error(err);
-                        return callback(null, new AccountInfo());
-                    });
-            },
-        );
+    public deleteAccount = async (
+        call: grpc.ServerUnaryCall<DeleteAccountRequest>,
+        callback: grpc.sendUnaryData<DeleteAccountResponse>,
+    ): Promise<void> => {
+        const accessToken = call.request.getAccessToken();
+        const response = new DeleteAccountResponse();
+        try {
+            const payload = await verifyToken(accessToken, config.accessToken.secret);
+            const account = await Account.findByPk(payload.sub);
+            if (!account) {
+                response.setError(DeleteAccountErrorStatus.DELETE_NOT_FOUND);
+            } else {
+                account.destroy();
```

Comment:
This may be async operation - it might be good to `await` it in order to handle possible failure